src/lasuite/malware_detection/backends/jcop.py

```diff
@@ -0,0 +1,268 @@
+        if not default_storage.exists(file_path):
+            raise FileNotFoundError(f"File {file_path} not found")
+
+        analyse_file_async.delay(
+            file_path,
+            api_key=self.api_key,
+            base_url=self.base_url,
+            callback_path=self.callback_path,
+            result_timeout=self.result_timeout,
+            submit_timeout=self.submit_timeout,
+            **kwargs,
+        )
+
+
+@shared_task(
+    bind=True,
+    default_retry_delay=30,
+    max_retries=10,
+    dont_autoretry_for=(MalwareDetectionInvalidAuthenticationError,),
+)
+def analyse_file_async(  # noqa: PLR0913
+    self,
+    file_path: str,
+    api_key: str,
+    base_url: str,
+    callback_path: str,
+    result_timeout: int = 30,
+    submit_timeout: int = 60 * 10,
+    file_hash: str | None = None,
+    **kwargs,
+) -> None:
+    """Task starting analysis process for a file."""
+    if not file_hash:
+        with default_storage.open(file_path, "rb") as file:
+            file_hash = hashlib.file_digest(file, "sha256").hexdigest()
+
+    callback = import_string(callback_path)
+
+    try:
+        # try if the file as already been tested
+        response = requests.get(
+            f"{base_url}/results/{file_hash}",
+            headers={
+                "X-Auth-Token": api_key,
+                "Accept": "application/json",
+            },
+            timeout=result_timeout,
+        )
+    except requests.exceptions.RequestException as exc:
+        logger.error("Error getting cache result for file %s: %s", file_path, exc)
+        if self.request.retries >= self.max_retries:
+            callback(
+                file_path,
+                ReportStatus.UNKNOWN,
+                error_info={
+                    "error": "Max retries fetching results exceeded",
+                    "error_code": 5000,
+                },
+                **kwargs,
+            )
+        self.retry(exc=exc)
+        return
+
+    if response.status_code == HTTPStatus.NOT_FOUND:
+        # start a new analysis
+        trigger_new_analysis.delay(
+            file_path,
+            api_key=api_key,
+            base_url=base_url,
+            callback_path=callback_path,
+            result_timeout=result_timeout,
+            submit_timeout=submit_timeout,
+            **kwargs,
+        )
+        return
+
+    if response.status_code == HTTPStatus.UNAUTHORIZED:
+        callback(
+            file_path,
+            ReportStatus.UNKNOWN,
+            error_info={
+                "error": "Invalid API key",
+                "error_code": response.status_code,
+            },
+            **kwargs,
+        )
+        raise MalwareDetectionInvalidAuthenticationError()
+
+    if response.status_code == HTTPStatus.OK:
+        content = response.json()
+        if content.get("done") is False:
+            # the analysis is not done yet, retry later
+            self.retry()
```

Comment:
I wonder if it's a good thing we manage "retries" at the task level for everything (network error, file not et analyzed, ...): in this specific case, do we need to wait the same amount of time? do we allow the same number of retries as when there is a timeout ?